src/webhook/webhook.service.ts

```diff
@@ -0,0 +1,99 @@
+import { HttpService } from '@nestjs/axios';
+import {
+  ForbiddenException,
+  Injectable,
+  NotFoundException,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { CreateWebhookRequest, FetchedWebhook, WebhookEvent } from './dto';
+import { WebhookRepository } from './webhook.repository';
+import { Recipe } from '@prisma/client';
+import Cryptr from 'cryptr';
+
+@Injectable()
+export class WebhookService {
+  constructor(
+    private readonly httpService: HttpService,
+    private readonly webhookRepository: WebhookRepository,
+  ) {}
+  cryptr = new Cryptr(process.env.CRYPTR_SECRET_KEY);
+
+  async createWebhook(
+    userId: number,
+    webhookData: CreateWebhookRequest,
+  ): Promise<void> {
+    const userWebhooks = await this.webhookRepository.getAllWebhooksByUserId(
+      userId,
+    );
+    if (userWebhooks.length >= +process.env.WEBHOOK_LIMIT) {
+      throw new ForbiddenException();
+    }
+
+    if (webhookData.token) {
+      const encryptedString = this.cryptr.encrypt(webhookData.token);
+      webhookData.token = encryptedString;
+    }
+
+    this.webhookRepository.createWebhook(userId, webhookData);
+  }
+
+  async deleteWebhook(userId: number, webhookId: number) {
+    const webhook = await this.webhookRepository.getWebhookById(webhookId);
+    if (!webhook) {
+      throw new NotFoundException();
+    }
+    if (userId !== webhook.userId) {
+      throw new UnauthorizedException();
+    }
+    this.webhookRepository.deleteUserWebhookById(webhookId);
```

Comment:
missing `await`? (unless we want to return the response without waiting the operation to finish? But I wouldn't do that in this case)